Assets/_FD/Script/GUI/MenuManager.cs

```diff
@@ -110,9 +115,37 @@ IEnumerator VictoryCo()
     {
         UI.SetActive(false);
         yield return new WaitForSeconds(1.5f);
-        VictotyUI.SetActive(true);
+        List<int> levelsDefined = new List<int>();
+        int currentLevel = GlobalValue.levelPlaying;
+        foreach (Reward reward in rewardList.rewards)
+        {
+            levelsDefined.Add(reward.rewardLevel);
+        }
+        
+        if (levelsDefined.Contains(currentLevel))
+            {
+                for (int i = 0; i < rewardList.rewards.Length; i++)
+                {
+                    if (currentLevel == rewardList.rewards[i].rewardLevel)
+                    {
+                        _currentReward = rewardList.rewards[i];
+                        rewardUI.SetActive(true);
+                        rewardUI.GetComponent<RewardMenu>().Init(_currentReward,this);
+                    }
+                }
+            }
+        else
+        {
+            VictotyUI.SetActive(true);    
+        }
```

Comment:
به جای اینکه دوتا حلقه استفاده کنی اینجا میتونی از Linq استفاده کنی
فقط باید اول کد این رو ایمپورت کنی:
using System.Linq;
```suggestion
        int currentLevel = GlobalValue.levelPlaying;
        var _currentReward =
            from item in rewardList.rewards
            where item.rewardLevel == currentLevel
            select item;
        foreach (Reward reward in _currentReward) {
            rewardUI.SetActive(true);
            rewardUI.GetComponent<RewardMenu>().Init(reward,this);
            yield break;
        }
        VictotyUI.SetActive(true);
```

Reply:
فقط حواست باشه این COmmint suggestion رو اگه بزنی اررور میگیره چون نیازه که using System.Linq; باشه حتما